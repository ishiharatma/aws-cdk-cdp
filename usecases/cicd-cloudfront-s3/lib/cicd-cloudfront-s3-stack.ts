import * as cdk from 'aws-cdk-lib';
import { StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { WebPipelineConstruct } from '../../common/lib/construct-pipeline-s3web'

import * as interfaceDef from '../../common/interface/index';

interface CicdCloudFrontS3StackProps extends StackProps {
  input: interfaceDef.WebPipelineConstructProps;
}

export class CicdCloudfrontS3Stack extends cdk.Stack {
  constructor(scope: Construct, id: string, props: CicdCloudFrontS3StackProps) {
    super(scope, id, props);

    new WebPipelineConstruct(this, 'Default', props.input);

  }
}
