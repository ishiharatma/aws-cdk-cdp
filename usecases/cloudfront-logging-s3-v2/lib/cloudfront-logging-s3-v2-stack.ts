import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import {
  aws_logs as logs,
} from 'aws-cdk-lib';
interface CloudFrontLoggingS3V2StackProps extends cdk.StackProps {
  readonly pjName: string;
  readonly envName: string;
  /*
    @param distributionId: string;
    @required
  */
    readonly distributionId: string;
    /*
      @param logPrefix?: string;
      @default /{DistributionId}/{yyyy}/{MM}/{dd}/{HH}
    */
    readonly logPrefix?: string;
    /*
      @param HiveCompatiblePath?: boolean;
      @default false
    */
    readonly HiveCompatiblePath?: boolean;
    /*
      @param logsDeliverySourceFormat?:string;
      see: https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutDeliveryDestination.html#API_PutDeliveryDestination_RequestSyntax
      @default json
    */
    readonly logsDeliveryDestinationFormat?:string;
    /*
      @param loggingBucketArn: string;
      @required
    */
    readonly loggingBucketArn: string;
}

export class CloudFrontLoggingS3V2Stack extends cdk.Stack {
  constructor(scope: Construct, id: string, props: CloudFrontLoggingS3V2StackProps) {
    super(scope, id, props);
    const accountId:string = cdk.Stack.of(this).account;
    const region:string = cdk.Stack.of(this).region;

    // logging v2
    // see: https://docs.aws.amazon.com/ja_jp/ja_jp/AmazonCloudFront/latest/DeveloperGuide/standard-logging.html
    // Logs: Delivery Source
    const logsDeliverySource = new logs.CfnDeliverySource(this, 'LogsDeliverySource', {
      name: `${props.pjName}-${props.envName}-cloudfront-delivery-source`,
      resourceArn: `arn:aws:cloudfront::${accountId}:distribution/${props.distributionId}`,
      logType: 'ACCESS_LOGS'
    });
    // Logs: Delivery Destination
    const logsDeliveryDestination = new logs.CfnDeliveryDestination(this, 'LogsDeliveryDestination', {
      name: `${props.pjName}-${props.envName}-cloudfront-delivery-destination`,
      destinationResourceArn: `${props.loggingBucketArn}`,///AWSLogs/${accountId}/CloudFront`,
      outputFormat: props.logsDeliveryDestinationFormat ?? 'json'
    });
    // Logs: Delivery
    const logsDelivery = new logs.CfnDelivery(this, 'LogsDelivery', {
      deliverySourceName: `${props.pjName}-${props.envName}-cloudfront-delivery-source`,
      deliveryDestinationArn: logsDeliveryDestination.attrArn,
      recordFields: [
        'timestamp', 'DistributionId', 'date', 'time', 'x-edge-location', 
        'sc-bytes', 'c-ip', 'cs-method', 'cs(Host)', 'cs-uri-stem', 
        'sc-status', 'cs(Referer)', 'cs(User-Agent)', 'cs-uri-query', 
        'cs(Cookie)', 'x-edge-result-type', 'x-edge-request-id', 
        'x-host-header', 'cs-protocol', 'cs-bytes', 'time-taken', 
        'x-forwarded-for', 'ssl-protocol', 'ssl-cipher', 
        'x-edge-response-result-type', 'cs-protocol-version', 'fle-status', 
        'fle-encrypted-fields', 'c-port', 'time-to-first-byte', 
        'x-edge-detailed-result-type', 'sc-content-type', 'sc-content-len', 
        'sc-range-start', 'sc-range-end', 'timestamp(ms)', 'origin-fbl', 
        'origin-lbl', 'asn'
      ],
      s3SuffixPath: props.logPrefix ?? '/v2/{DistributionId}/{yyyy}/{MM}/{dd}/{HH}',
      s3EnableHiveCompatiblePath: props.HiveCompatiblePath ?? false
    });
    // 依存関係の設定
    logsDelivery.addDependency(logsDeliverySource);
    logsDelivery.addDependency(logsDeliveryDestination);

  }
}
