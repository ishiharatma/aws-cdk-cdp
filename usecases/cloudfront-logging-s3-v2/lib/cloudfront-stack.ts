import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import {
  aws_s3 as s3,
  aws_cloudfront as cloudfront,
  aws_cloudfront_origins as origins, 
  aws_s3_deployment as s3deploy,
  aws_logs as logs,
} from 'aws-cdk-lib';
import * as path from 'path';
interface CloudFrontStackProps extends cdk.StackProps {
  readonly pjName: string;
  readonly envName: string;
  readonly isAutoDeleteObject?: boolean;
}

export class CloudFrontStack extends cdk.Stack {
  public readonly distribution:cloudfront.IDistribution;
  public readonly loggingBucket:s3.IBucket;
  constructor(scope: Construct, id: string, props: CloudFrontStackProps) {
    super(scope, id, props);
    const accountId:string = cdk.Stack.of(this).account;
    const region:string = cdk.Stack.of(this).region;
    const contentsPath:string = path.join(__dirname, '../src/static-site');

    // create logging bucket for cloudfront
    this.loggingBucket = new s3.Bucket(this, 'LoggingBucket', {
      removalPolicy: cdk.RemovalPolicy.DESTROY,
      autoDeleteObjects: true,
    }); 
    // create static website bucket
    const staticWebSiteBucket = new s3.Bucket(this, 'StaticWebSiteBucket', {
      removalPolicy: cdk.RemovalPolicy.DESTROY,
      autoDeleteObjects: true,
    });

    const responseHeadersPolicy = new cloudfront.ResponseHeadersPolicy(
      this,
      'ResponseHeadersPolicy',
      {
        securityHeadersBehavior: {
          contentTypeOptions: { override: true },
          frameOptions: {
            frameOption: cloudfront.HeadersFrameOption.DENY,
            override: true,
          },
          referrerPolicy: {
            referrerPolicy: cloudfront.HeadersReferrerPolicy.SAME_ORIGIN,
            override: true,
          },
          strictTransportSecurity: {
            accessControlMaxAge: cdk.Duration.seconds(63072000),
            includeSubdomains: true,
            preload: true,
            override: true,
          },
          xssProtection: {
            protection: true,
            modeBlock: true,
            override: true,
          },
        },
        customHeadersBehavior: {
          customHeaders: [
            {
              header: 'Cache-Control',
              value: 'no-cache',
              override: true,
            },
            {
              header: 'pragma',
              value: 'no-cache',
              override: true,
            },
            {
              header: 'server',
              value: '',
              override: true,
            },
          ],
        },
      }
    );
    // create cloudfront distribution
    this.distribution = new cloudfront.Distribution(this, 'MyDistribution', {
      comment: 'My distribution',
      enabled: true,
      httpVersion: cloudfront.HttpVersion.HTTP2_AND_3,
      // see: https://docs.aws.amazon.com/ja_jp/AmazonCloudFront/latest/DeveloperGuide/PriceClass.html
      priceClass: cloudfront.PriceClass.PRICE_CLASS_200, //or PRICE_CLASS_ALL
      minimumProtocolVersion: cloudfront.SecurityPolicyProtocol.TLS_V1_2_2021,
      enableIpv6: false,
      defaultRootObject: "index.html",

      defaultBehavior: {
        allowedMethods: cloudfront.AllowedMethods.ALLOW_GET_HEAD_OPTIONS,
        cachedMethods: cloudfront.CachedMethods.CACHE_GET_HEAD,
        cachePolicy: cloudfront.CachePolicy.CACHING_OPTIMIZED,
        viewerProtocolPolicy: cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
        compress: true,
        origin: origins.S3BucketOrigin.withOriginAccessControl(staticWebSiteBucket),
        originRequestPolicy: cloudfront.OriginRequestPolicy.CORS_S3_ORIGIN,
        responseHeadersPolicy
      },
    });

    new s3deploy.BucketDeployment(this, 'IndexHtmlDeploy', {
      sources: [
        s3deploy.Source.asset(contentsPath),
      ],
      destinationBucket: staticWebSiteBucket,
      distribution: this.distribution,
      distributionPaths: ["/*"],
    });

  }
}
