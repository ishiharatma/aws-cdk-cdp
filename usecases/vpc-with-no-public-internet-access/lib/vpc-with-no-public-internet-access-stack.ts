
import { Stack, StackProps } from 'aws-cdk-lib';
import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { 
  aws_s3 as s3,
  aws_ec2 as ec2, 
  aws_iam as iam,
  aws_kms as kms,
} from 'aws-cdk-lib';

interface MyVpcStackProps extends StackProps {
  readonly pjName: string;
  readonly envName: string;
  readonly vpcCIDR: string
  /**
   * Define the maximum number of AZs to use in this region
   * @default 2 AZ
   */
  readonly maxAzs?: number;
  readonly isAutoDeleteObject: boolean;
} 

export class VpcWithNoPublicInternetAccessStack extends cdk.Stack {
  public readonly vpc: ec2.IVpc;
  constructor(scope: Construct, id: string, props: MyVpcStackProps) {
    super(scope, id, props);

    const accountId:string = cdk.Stack.of(this).account;
    const region:string = cdk.Stack.of(this).region;
    // VPC
    this.vpc = new ec2.Vpc(this, 'MyVpc', {
      vpcName: [props.pjName, props.envName, 'VPC', accountId].join('/') ,
      ipAddresses: ec2.IpAddresses.cidr(props.vpcCIDR),
      maxAzs: props.maxAzs ?? 2, // 2 Availability Zones
      createInternetGateway: false, // No Internet Gateway
      subnetConfiguration: [
        {
          cidrMask: 24,
          name: 'PrivateSubnet',
          subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS,
        },
        {
          cidrMask: 24,
          name: 'IsolatedSubnet',
          subnetType: ec2.SubnetType.PRIVATE_ISOLATED,
        },
      ],
      natGateways: 0, // No NAT Gateway
      // Security Hub EC2.2
      // https://docs.aws.amazon.com/ja_jp/securityhub/latest/userguide/ec2-controls.html#ec2-2
      restrictDefaultSecurityGroup: true, 
    });

    // Security Hub EC2.6
    // https://docs.aws.amazon.com/ja_jp/securityhub/latest/userguide/ec2-controls.html#ec2-6
    // FlowLog
    // (Optional)CMK
    /*
    const flowLogKey = new kms.Key(this, 'Key', {
      enableKeyRotation: true,
      description: 'for VPC Flow log',
      alias: `${props.pjName}-${props.envName}-for-flowlog`,
    });
    new cdk.CfnOutput(this, 'KMSKeyId', {
      value: flowLogKey.keyId,
    });
    flowLogKey.addToResourcePolicy(
      new iam.PolicyStatement({
        actions: ['kms:Encrypt*', 'kms:Decrypt*', 'kms:ReEncrypt*', 'kms:GenerateDataKey*', 'kms:Describe*'],
        principals: [new iam.ServicePrincipal('delivery.logs.amazonaws.com')],
        resources: ['*'],
      }),
    );
    */
    // S3 Bucket for FlowLogs
    const flowLogsBucket  = new s3.Bucket(this, 'FlowLogsBucket', {
      bucketName: [props.pjName, props.envName,'flowlogs', accountId].join('-') ,
      accessControl: s3.BucketAccessControl.PRIVATE,
      blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
      objectOwnership: s3.ObjectOwnership.BUCKET_OWNER_ENFORCED,
      encryption: s3.BucketEncryption.S3_MANAGED,
      // Optional if you want to use CMK
      //encryption: s3.BucketEncryption.KMS,
      //encryptionKey: flowLogKey,
      removalPolicy: props.isAutoDeleteObject ? cdk.RemovalPolicy.DESTROY: cdk.RemovalPolicy.RETAIN, // Delete if isAutoDeleteObject is true, otherwise do not delete
      autoDeleteObjects: props.isAutoDeleteObject,
    });
    new cdk.CfnOutput(this, 'FlowLogsBucketName', {
      value: flowLogsBucket.bucketName,
    });

    this.vpc.addFlowLog('FlowLogs', {
      destination: ec2.FlowLogDestination.toS3(flowLogsBucket),
      trafficType: ec2.FlowLogTrafficType.ALL,
    });

    // Gateway 型 VPC エンドポイント
    // VPC Endpoint for S3
    this.vpc.addGatewayEndpoint('S3VPCE',{
      service: ec2.GatewayVpcEndpointAwsService.S3,
      subnets: [{ subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS }], // (CDK v2.69 deprecated)PRIVATE_WITH_NAT
    });

  }
}

