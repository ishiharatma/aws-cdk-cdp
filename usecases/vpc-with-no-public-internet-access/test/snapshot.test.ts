import { App, Stack } from 'aws-cdk-lib';
import { Match, Template } from 'aws-cdk-lib/assertions';
import { VpcWithNoPublicInternetAccessStack } from '../lib/vpc-with-no-public-internet-access-stack';

const pjName = 'snapshot';
const envName = 'test';

const defaultEnv = {
    account: '123456789012',
    region: 'ap-northeast-1',
};

test('Snapshot', () => {
    // GIVEN
    const app = new App({
        context : {}
    });
    const stack = new VpcWithNoPublicInternetAccessStack(app, 'CloudFrontS3OacStack', {
        pjName: pjName,
        envName: envName,
        vpcCIDR: '10.0.0.0/16',
        isAutoDeleteObject: true,
        description: 'VPC with no public internet access',
        env: defaultEnv,
        terminationProtection: true, // Enab
      });
      
    // WHEN
    const template = Template.fromStack(stack);
    // THEN
    // test with snapshot
    expect(template).toMatchSnapshot();

});
