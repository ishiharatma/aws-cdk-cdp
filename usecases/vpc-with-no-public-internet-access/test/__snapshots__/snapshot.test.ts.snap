// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
{
  "Description": "VPC with no public internet access",
  "Outputs": {
    "FlowLogsBucketName": {
      "Value": {
        "Ref": "FlowLogsBucket8912DCDB",
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F": {
      "DependsOn": [
        "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": "cdk-hnb659fds-assets-123456789012-ap-northeast-1",
          "S3Key": "faa95a81ae7d7373f3e1f242268f904eb748d8d0fdd306e8a6fe515a1905a7d6.zip",
        },
        "Description": {
          "Fn::Join": [
            "",
            [
              "Lambda function for auto-deleting objects in ",
              {
                "Ref": "FlowLogsBucket8912DCDB",
              },
              " S3 bucket.",
            ],
          ],
        },
        "Handler": "index.handler",
        "MemorySize": 128,
        "Role": {
          "Fn::GetAtt": [
            "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
            "Arn",
          ],
        },
        "Runtime": "nodejs20.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CustomVpcRestrictDefaultSGCustomResourceProviderHandlerDC833E5E": {
      "DependsOn": [
        "CustomVpcRestrictDefaultSGCustomResourceProviderRole26592FE0",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": "cdk-hnb659fds-assets-123456789012-ap-northeast-1",
          "S3Key": "7fa1e366ee8a9ded01fc355f704cff92bfd179574e6f9cfee800a3541df1b200.zip",
        },
        "Description": "Lambda function for removing all inbound/outbound rules from the VPC default security group",
        "Handler": "__entrypoint__.handler",
        "MemorySize": 128,
        "Role": {
          "Fn::GetAtt": [
            "CustomVpcRestrictDefaultSGCustomResourceProviderRole26592FE0",
            "Arn",
          ],
        },
        "Runtime": "nodejs20.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomVpcRestrictDefaultSGCustomResourceProviderRole26592FE0": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
        "Policies": [
          {
            "PolicyDocument": {
              "Statement": [
                {
                  "Action": [
                    "ec2:AuthorizeSecurityGroupIngress",
                    "ec2:AuthorizeSecurityGroupEgress",
                    "ec2:RevokeSecurityGroupIngress",
                    "ec2:RevokeSecurityGroupEgress",
                  ],
                  "Effect": "Allow",
                  "Resource": [
                    {
                      "Fn::Join": [
                        "",
                        [
                          "arn:",
                          {
                            "Ref": "AWS::Partition",
                          },
                          ":ec2:ap-northeast-1:123456789012:security-group/",
                          {
                            "Fn::GetAtt": [
                              "MyVpcF9F0CA6F",
                              "DefaultSecurityGroup",
                            ],
                          },
                        ],
                      ],
                    },
                  ],
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "Inline",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "FlowLogsBucket8912DCDB": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "AccessControl": "Private",
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "BucketName": "snapshot-test-flowlogs-123456789012",
        "OwnershipControls": {
          "Rules": [
            {
              "ObjectOwnership": "BucketOwnerEnforced",
            },
          ],
        },
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
        "Tags": [
          {
            "Key": "aws-cdk:auto-delete-objects",
            "Value": "true",
          },
        ],
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "FlowLogsBucketAutoDeleteObjectsCustomResource261C6D2B": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "FlowLogsBucketPolicy49FA954A",
      ],
      "Properties": {
        "BucketName": {
          "Ref": "FlowLogsBucket8912DCDB",
        },
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomS3AutoDeleteObjectsCustomResourceProviderHandler9D90184F",
            "Arn",
          ],
        },
      },
      "Type": "Custom::S3AutoDeleteObjects",
      "UpdateReplacePolicy": "Delete",
    },
    "FlowLogsBucketPolicy49FA954A": {
      "Properties": {
        "Bucket": {
          "Ref": "FlowLogsBucket8912DCDB",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:PutBucketPolicy",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
              ],
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::GetAtt": [
                    "CustomS3AutoDeleteObjectsCustomResourceProviderRole3B1BD092",
                    "Arn",
                  ],
                },
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "FlowLogsBucket8912DCDB",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "FlowLogsBucket8912DCDB",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "MyVpcF9F0CA6F": {
      "Properties": {
        "CidrBlock": "10.0.0.0/16",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": [
          {
            "Key": "Name",
            "Value": "snapshot/test/VPC/123456789012",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
    "MyVpcFlowLogsFlowLogF83C1C7A": {
      "DependsOn": [
        "FlowLogsBucketAutoDeleteObjectsCustomResource261C6D2B",
        "FlowLogsBucketPolicy49FA954A",
      ],
      "Properties": {
        "LogDestination": {
          "Fn::GetAtt": [
            "FlowLogsBucket8912DCDB",
            "Arn",
          ],
        },
        "LogDestinationType": "s3",
        "ResourceId": {
          "Ref": "MyVpcF9F0CA6F",
        },
        "ResourceType": "VPC",
        "Tags": [
          {
            "Key": "Name",
            "Value": "CloudFrontS3OacStack/MyVpc/FlowLogs",
          },
        ],
        "TrafficType": "ALL",
      },
      "Type": "AWS::EC2::FlowLog",
    },
    "MyVpcIsolatedSubnetSubnet1RouteTableAssociation4F6F5B7B": {
      "Properties": {
        "RouteTableId": {
          "Ref": "MyVpcIsolatedSubnetSubnet1RouteTableDE4E2393",
        },
        "SubnetId": {
          "Ref": "MyVpcIsolatedSubnetSubnet1SubnetBEB69234",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "MyVpcIsolatedSubnetSubnet1RouteTableDE4E2393": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "CloudFrontS3OacStack/MyVpc/IsolatedSubnetSubnet1",
          },
        ],
        "VpcId": {
          "Ref": "MyVpcF9F0CA6F",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "MyVpcIsolatedSubnetSubnet1SubnetBEB69234": {
      "Properties": {
        "AvailabilityZone": "dummy1a",
        "CidrBlock": "10.0.2.0/24",
        "MapPublicIpOnLaunch": false,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "IsolatedSubnet",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Isolated",
          },
          {
            "Key": "Name",
            "Value": "CloudFrontS3OacStack/MyVpc/IsolatedSubnetSubnet1",
          },
        ],
        "VpcId": {
          "Ref": "MyVpcF9F0CA6F",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "MyVpcIsolatedSubnetSubnet2RouteTable9E8E775E": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "CloudFrontS3OacStack/MyVpc/IsolatedSubnetSubnet2",
          },
        ],
        "VpcId": {
          "Ref": "MyVpcF9F0CA6F",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "MyVpcIsolatedSubnetSubnet2RouteTableAssociation0542B7DC": {
      "Properties": {
        "RouteTableId": {
          "Ref": "MyVpcIsolatedSubnetSubnet2RouteTable9E8E775E",
        },
        "SubnetId": {
          "Ref": "MyVpcIsolatedSubnetSubnet2Subnet4DC0B981",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "MyVpcIsolatedSubnetSubnet2Subnet4DC0B981": {
      "Properties": {
        "AvailabilityZone": "dummy1b",
        "CidrBlock": "10.0.3.0/24",
        "MapPublicIpOnLaunch": false,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "IsolatedSubnet",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Isolated",
          },
          {
            "Key": "Name",
            "Value": "CloudFrontS3OacStack/MyVpc/IsolatedSubnetSubnet2",
          },
        ],
        "VpcId": {
          "Ref": "MyVpcF9F0CA6F",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "MyVpcPrivateSubnetSubnet1RouteTableAssociation2811D7AF": {
      "Properties": {
        "RouteTableId": {
          "Ref": "MyVpcPrivateSubnetSubnet1RouteTableD918A34F",
        },
        "SubnetId": {
          "Ref": "MyVpcPrivateSubnetSubnet1SubnetE8BD536C",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "MyVpcPrivateSubnetSubnet1RouteTableD918A34F": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "CloudFrontS3OacStack/MyVpc/PrivateSubnetSubnet1",
          },
        ],
        "VpcId": {
          "Ref": "MyVpcF9F0CA6F",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "MyVpcPrivateSubnetSubnet1SubnetE8BD536C": {
      "Properties": {
        "AvailabilityZone": "dummy1a",
        "CidrBlock": "10.0.0.0/24",
        "MapPublicIpOnLaunch": false,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "PrivateSubnet",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          {
            "Key": "Name",
            "Value": "CloudFrontS3OacStack/MyVpc/PrivateSubnetSubnet1",
          },
        ],
        "VpcId": {
          "Ref": "MyVpcF9F0CA6F",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "MyVpcPrivateSubnetSubnet2RouteTable83C86AF7": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "CloudFrontS3OacStack/MyVpc/PrivateSubnetSubnet2",
          },
        ],
        "VpcId": {
          "Ref": "MyVpcF9F0CA6F",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "MyVpcPrivateSubnetSubnet2RouteTableAssociationE7A10A88": {
      "Properties": {
        "RouteTableId": {
          "Ref": "MyVpcPrivateSubnetSubnet2RouteTable83C86AF7",
        },
        "SubnetId": {
          "Ref": "MyVpcPrivateSubnetSubnet2SubnetE3BFCF91",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "MyVpcPrivateSubnetSubnet2SubnetE3BFCF91": {
      "Properties": {
        "AvailabilityZone": "dummy1b",
        "CidrBlock": "10.0.1.0/24",
        "MapPublicIpOnLaunch": false,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "PrivateSubnet",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          {
            "Key": "Name",
            "Value": "CloudFrontS3OacStack/MyVpc/PrivateSubnetSubnet2",
          },
        ],
        "VpcId": {
          "Ref": "MyVpcF9F0CA6F",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "MyVpcRestrictDefaultSecurityGroupCustomResourceA4FCCD62": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "Account": "123456789012",
        "DefaultSecurityGroupId": {
          "Fn::GetAtt": [
            "MyVpcF9F0CA6F",
            "DefaultSecurityGroup",
          ],
        },
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomVpcRestrictDefaultSGCustomResourceProviderHandlerDC833E5E",
            "Arn",
          ],
        },
      },
      "Type": "Custom::VpcRestrictDefaultSG",
      "UpdateReplacePolicy": "Delete",
    },
    "MyVpcS3VPCE1909106E": {
      "Properties": {
        "RouteTableIds": [
          {
            "Ref": "MyVpcPrivateSubnetSubnet1RouteTableD918A34F",
          },
          {
            "Ref": "MyVpcPrivateSubnetSubnet2RouteTable83C86AF7",
          },
        ],
        "ServiceName": {
          "Fn::Join": [
            "",
            [
              "com.amazonaws.",
              {
                "Ref": "AWS::Region",
              },
              ".s3",
            ],
          ],
        },
        "VpcEndpointType": "Gateway",
        "VpcId": {
          "Ref": "MyVpcF9F0CA6F",
        },
      },
      "Type": "AWS::EC2::VPCEndpoint",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
